### ISSUE TITLE
[Help needed] OpenCL issues when creating Ubuntu 22.04 base docker image

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#1775]
**[Summary]**:
When I am trying to build OpenCL backend (static check only) with `ci_rocm530_ubuntu2204` branch on the `rocm/miopen:ci_rocm530_ubuntu2204` docker, I am seeing the following strange errors:
```
[  0%] Building CXX object src/CMakeFiles/MIOpen.dir/__/kernel.cpp.o
c++: error: unrecognized command-line option ‘-mllvm’
c++: error: unrecognized command-line option ‘-amdgpu-early-inline-all=true’
c++: error: unrecognized command-line option ‘-mllvm’
c++: error: unrecognized command-line option ‘-amdgpu-function-calls=false’
c++: error: unrecognized command-line option ‘--offload-arch=gfx900’
c++: error: unrecognized command-line option ‘--offload-arch=gfx906’
c++: error: unrecognized command-line option ‘--offload-arch=gfx908’
c++: error: unrecognized command-line option ‘--offload-arch=gfx90a’
c++: error: unrecognized command-line option ‘--offload-arch=gfx1030’
```

**[Steps to reproduce]**:

1. Pull the docker `rocm/miopen:ci_rocm530_ubuntu2204` 
2. Clone MIOpen:
```
export LC_ALL=C.UTF-8 && \
export LANG=C.UTF-8 && \
sudo apt-get update -y && \
sudo apt-get -y install git unzip python3-venv pkg-config && \
sudo apt update && \
cd /root && git clone -b develop https://github.com/ROCmSoftwarePlatform/MIOpen && cd MIOpen
```
3. Checkout the branch: `ci_rocm530_ubuntu2204`
4. The following commands:
```
cmake -DMIOPEN_BACKEND=OpenCL -DCMAKE_PREFIX_PATH=/opt/rocm -DBUILD_DEV=On .. && make -j$(nproc)
``` 
example:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/ci_rocm530_ubuntu2204/9/pipeline/14

**[Discussions]**:
1. Why is `c++` selected when `clang++` should be compiling with those special command-line options?

**Comment by atamazov** [comment]
@junliume It seems like cmake finds some old clang++ somewhere in the docker and uses it for building the host code. Does adding `CXX=g++` prior `cmake...` resolve the issue?

**Comment by junliume** [comment]
> CXX=g++

No, adding `CXX=g++` does not resolve the issue, unfortunately. And this PR has the same issue:
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1761

This above PR might be the root cause of the issue though, since I have merged it into this ci_rocm530_ubuntu2204 branch

**Comment by pfultz2** [comment]
Try adding `VERBOSE=1` to the make command so it will print out how its calling the compiler.

It looks like `hip::device` is being linked on OpenCL backend. Maybe this is coming indirectly from another dependency that libMIOpen is using, such as comgr, mlir, or CK. Also it could be something unusual is happening with `find_package(hip)` as well.

**Comment by junliume** [comment]
> Try adding `VERBOSE=1` to the make command so it will print out how its calling the compiler.
> 
> It looks like `hip::device` is being linked on OpenCL backend. Maybe this is coming indirectly from another dependency that libMIOpen is using, such as comgr, mlir, or CK. Also it could be something unusual is happening with `find_package(hip)` as well.

Thanks, @pfultz2 !  PR https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1761 might be the cause and libMIOpen using CK could be the prime suspect. Just not sure how at this moment.

**Comment by junliume** [comment]
@atamazov  @pfultz2 I might be missing an obvious one: CK does not support OpenCL backend.
@JehandadKhan  we need to isolate CK when OpenCL backend is chosen ...

**Comment by pfultz2** [comment]
> I might be missing an obvious one: CK does not support OpenCL backend.

Yea I am thinking the same thing, but it seems to be unconditionally enabled though.

**Comment by junliume** [comment]
Resolved by #1761
